traitlets/config/configurable.py

```diff
@@ -308,16 +345,29 @@ def c(s):
             lines.append(c(desc))
             lines.append('')
 
-        for name, trait in sorted(cls.class_own_traits(config=True).items()):
-            lines.append(c(trait.help))
+        for name, trait in sorted(cls.class_traits(config=True).items()):
+            default_repr = trait.default_value_repr()
 
-            if 'Enum' in type(trait).__name__:
-                # include Enum choices
-                lines.append('#  Choices: %r' % (trait.values,))
+            if classes:
+                defining_class = cls._defining_class(trait, classes)
+            else:
+                defining_class = cls
+            if defining_class is cls:
+                # cls owns the trait, show full help
+                if trait.help:
+                    lines.append(c(trait.help))
+                if 'Enum' in type(trait).__name__:
+                    # include Enum choices
+                    lines.append('#  Choices: %r' % (trait.values,))
+                lines.append('#  Default: %s' % default_repr)
+            else:
+                # Trait appears multiple times and isn't defined here.
+                # Truncate help to first line + "See also Original.trait"
+                if trait.help:
+                    lines.append(c(trait.help.split('\n', 1)[0]))
+                lines.append('#  See also %s.%s' % (defining_class.__name__, name))
```

Comment:
I would put a `:` at the end of the label to look like a "field", like `"#  Default:"` and `"#  Choices:"`.